ethcore/src/client/client.rs

```diff
@@ -1240,6 +1241,8 @@ impl Client {
 
 			let mut ret = Executive::new(state, env_info, machine).transact_virtual(transaction, options)?;
 
+			ret.transaction_hash = if hash_trace { Some(transaction.hash()) } else { None };
```

Comment:
tbh, I don't know what to do with this. I generally don't like mutating structures like this. Making one field optional and then overwriting it in a different function is not idiomatic.

Unfortunately, it's not a new pattern and the same thing is done with `ret.state_diff` few lines below :/

Reply:
I don't know what can be done with this since it depends on if the user wants to trace transaction hashes or not.
Another way can be implementing it the same way but inside `replay_block_transactions` but I thought implementing this inside the same function that `state_diff` is implemented would look better.

Reply:
Maybe I can create a function that returns transaction hash inside the implementation of the `Executed` Struct like this:

`impl Executed {
	pub fn transaction_hash(transaction: &SignedTransaction) -> Option<H256> {
		...........................
	}
}`

then inside the do_virtual_call function I would set transaction_hash with this:
`if hash_trace { Some(ret.transaction_hash(transaction) };`

what do you think @debris